MSCI445-Lab2/EmailLab/MainWindow.xaml.cs

```diff
@@ -39,21 +40,47 @@ public MainWindow()
 
 		private void Submit_Button_Click(object sender, RoutedEventArgs e)
         {
-			validateInput();
 			myEmail = new MyEmail(SUBJECT_TEXTBOX.Text, 
 				MESSAGE_TEXTBOX.Text, FROM_TEXTBOX.Text, "", TO_TEXTBOX.Text);
 			var passwordWindow = new PasswordWindow(myEmail);
-			passwordWindow.Show();
+			if (this.validateInput() == true) { passwordWindow.Show(); };
 		}
 
-        private void validateInput()
+		private bool validateInput()
         {
+			bool stats = true;
+			String errors = "One or more errors occured: \n";
+			try
+			{
+				if (FROM_TEXTBOX.Text != "") { MailAddress s = new MailAddress(FROM_TEXTBOX.Text); } else { errors += "Sender's email address can not be empty. \n"; }
```

Comment:
we can do the MailAddress initialization in `MyEmail` class. For this method, we should simply check the string, not initialize anything.